src/main/java/paymelah/model/debt/DebtList.java

```diff
@@ -82,6 +82,42 @@ public DebtList removeDebt(Debt toRemove) {
         return edited;
     }
 
+    /**
+     * Marks a debt from the list as paid.
+     *
+     * @param toMark The debt to mark as paid from the list.
+     * @return The modified {@code DebtList}.
+     */
+    public DebtList markDebt(Debt toMark) {
+        requireNonNull(toMark);
+        if (!contains(toMark)) {
+            throw new DebtNotFoundException();
+        }
+        DebtList edited = new DebtList(this);
+        edited.debts.remove(toMark);
+        edited.debts.add(toMark.setPaid(true));
+        edited.totalDebt = totalDebt.subtract(toMark.getMoney().getValue());
+        return edited;
+    }
+
+    /**
+     * Marks a debt from the list as unpaid.
+     *
+     * @param toUnmark The debt to mark as unpaid from the list.
+     * @return The modified {@code DebtList}.
+     */
+    public DebtList unmarkDebt(Debt toUnmark) {
+        requireNonNull(toUnmark);
+        if (!contains(toUnmark)) {
+            throw new DebtNotFoundException();
+        }
+        DebtList edited = new DebtList(this);
+        edited.debts.remove(toUnmark);
+        edited.debts.add(toUnmark.setPaid(false));
+        edited.totalDebt = totalDebt.add(toUnmark.getMoney().getValue());
+        return edited;
+    }
+
```

Comment:
Might need to add a check for if `Debt` is already marked/unmarked before making changes to `totalDebt`.